### PR TITLE
Minor improvement, fix some bugs

### PR DESCRIPTION
**Pull request by Gnouc**
Some minor improvements:

 - Replace useless use of `dirname` with standard `${var%/*}`.

 - Fix missing double quotes

 - Make some construct work with standard POSIX shell:

  + Using `=` instead of `==` for old test `[...]`
  + Using `:` for no-op operator instead of empty `if` body, even in `bash`, the empty `if` body will raise an error

PS: And also many places used ` 2>&1 >/dev/null`, it does not make sense, should be `>/dev/null 2>&1` if you want to discard all output.

### CODE REVIEW

**Comment by Gnouc** [comment]
@maddox Does anyone look at this?

**Comment by technicalpickles** [comment]
@Gnouc yes, but things get pretty slow in the second half of December.

**Comment by mikemcquaid** [comment]
I'd probably just say we update to use `/bin/bash` instead.

**Comment by Gnouc** [comment]
@mikemcquaid IMHO, update to use `/bin/bash` make no sense. It's only allow you to use `==` inside old test `[...]`. The empty `if` body won't work with `bash`.

**Comment by mikemcquaid** [comment]
@Gnouc Most of the scripts we use in GitHub have bashisms and these do too. We only run these on systems where there is a `/bin/bash`. I've personally dabbled in avoiding Bashisms in the past and unless you're really wanting to support e.g. Solaris I don't think it's worth the effort. There's still other stuff worth having in this PR, though :+1:

**Comment by Gnouc** [comment]
@mikemcquaid Yes, I personally do not against bashisms (although I have created [a tool do it](https://github.com/Gnouc/flycheck-checkbashisms) :stuck_out_tongue: ) but we must be consistent between the shebang and our shell script syntax, right :relaxed:

Anyway, FWIW, you can read [this link](http://unix.stackexchange.com/a/148061/38906) for some benchmark pointing out how slow `bash` is.

**Comment by mikemcquaid** [comment]
@Gnouc I agree and think the shebangs should be what's updated, rather than the Bashisms.

**Comment by maddox** [comment]
It's also important to remember that all of these scripts are mostly a **reference** to the STRTA pattern, rather than actually used scripts.

**Comment by Gnouc** [comment]
So, should I make the shebang change to `#!/bin/bash` in this pull or you will make?

**Comment by mikemcquaid** [comment]
> So, should I make the shebang change to #!/bin/bash in this pull

That'd be great, thanks :+1:

**Comment by Gnouc** [comment]
@maddox @mikemcquaid Should it be merged now?

File: script/bootstrap [hunk 1]
```diff
@@ -1,16 +1,16 @@
-#!/bin/sh
+#!/bin/bash
 
 # script/bootstrap: Resolve all dependencies that the application requires to
 #                   run.
 
 set -e
 
-cd "$(dirname "$0")/.."
+cd "${0%/*}/.."
```
Comment (mikemcquaid) [inline script/bootstrap]: I think it's clearer the way it was before, here.

File: script/bootstrap [hunk 2]
```diff
@@ -1,16 +1,16 @@
...
 
 if [ -f "Brewfile" ] && [ "$(uname -s)" = "Darwin" ]; then
   brew update
 
-  brew bundle check 2>&1 >/dev/null || {
+  brew bundle check >/dev/null 2>&1 || {
```
Comment (mikemcquaid) [inline script/bootstrap]: What advantage do you get from flipping the ordering here? Also, given we're using `bash` anyway I wonder if we should just go for `&>/dev/null`.

File: script/setup [hunk 3]
```diff
@@ -18,6 +18,7 @@ if [ -z "$RAILS_ENV" ] && [ -z "$RACK_ENV" ]; then
   # Do things that need to be done to the application to set up for the first
   # time. Or things needed to be run to to reset the application back to first
   # use experience. These things are scoped to the application's domain.
+  :
```
Comment (mikemcquaid) [inline script/setup]: What's this for, out of interest?

File: script/bootstrap (hunk 2 above)
Comment (Gnouc) [inline script/bootstrap]: `2>&1 >/dev/null` is the wrong and useless use. You still get the error message (if any) in terminal. Try:
```
ls file_does_not_exist 2>&1 >/dev/null
```

File: script/bootstrap (hunk 1 above)
Comment (Gnouc) [inline script/bootstrap]: I'm not sure, IMHO, parameter expansion is the right and idiom way to do this task.

See also [my answer](http://unix.stackexchange.com/a/254910/38906) and the link I gave there for more details.

File: script/bootstrap (hunk 2 above)
Comment (mikemcquaid) [inline script/bootstrap]: Thanks. Thoughts on `&>/dev/null`?

File: script/bootstrap (hunk 1 above)
Comment (mikemcquaid) [inline script/bootstrap]: Sure; I do think it's a bit less obvious though to non-Bash experts.

File: script/setup (hunk 3 above)
Comment (Gnouc) [inline script/setup]: What do you mean?
```
if [ 1 = 1 ]; then
  # Comment
fi
```
give you syntax error.

File: script/bootstrap (hunk 2 above)
Comment (Gnouc) [inline script/bootstrap]: Since `&>/dev/null` does the same task, I chose `>/dev/null 2>&1` for readability.

**Comment by Gnouc** [comment]
Well, after some conversation, I saw we have some conflicts in coding style. Since when there's no coding convention for this repo, it's hard for me to chose which one to change. So I close this pull.

Thanks all for the discussion.

File: script/setup (hunk 3 above)
Comment (mikemcquaid) [inline script/setup]: Ok, thanks. What does `:` do by itself? I'm interested in learning :grinning:

Comment (Gnouc) [inline script/setup]: It's [null command](http://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_16), do nothing, like no-op operator.

**Pull request closed without merging**